premake/generator.lua

```diff
@@ -284,6 +284,16 @@ function CreateProject(config)
 		end
 	end
 
+	local pch_enabled = false
+	if config.pch_header ~= nil or config.pch_source ~= nil then
+		assert(config.pch_header ~= nil, "'phc_header' must be supplied when 'pch_source' is supplied!")
+		assert(config.pch_source ~= nil, "'pch_source' must be supplied when 'phc_header' is supplied!")
+		assert(type(config.pch_header) == "string", "'pch_header' is not a string!")		
+		assert(type(config.pch_source) == "string", "'pch_source' is not a string!")
+		assert(os.isfile(sourcepath .. "/" .. config.pch_source), "'pch_source' file " .. config.pch_source .. " could not be found!")
```

Comment:
Why is there no file path validation for the header too?

Reply:
Refer to the comment above; we can assume where pch_source will be located, but we can't do the same for pch_header since it may relative to the source directory or an include directory.